builtin/providers/aws/data_source_aws_ami.go

```diff
@@ -236,10 +236,19 @@ func dataSourceAwsAmiRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	var filteredImages []*ec2.Image
-	if nameRegexOk == true {
+	if nameRegexOk {
 		r := regexp.MustCompile(nameRegex.(string))
 		for _, image := range resp.Images {
-			if r.MatchString(*image.Name) == true {
+			// Check for a very rare case where the response would include no
+			// image name. No name means nothing to attempt a match against,
+			// therefore we are skipping such image.
+			if image.Name == nil || *image.Name == "" {
+				log.Printf("[WARN] Unable to find AMI name to match against "+
+					"for image ID %q owned by %q, nothing to do.",
+					*image.ImageId, *image.OwnerId)
```

Comment:
A little bit of healthy paranoia is what makes you question things, which is often good. 👍 

You need to have an account to be able to create and/or upload an image to use later, plus this is provided by the backend, nothing that user can control and/or set. In other words, I believe we are fine with both image ID and owner.